packages/create-amplify/src/logger.ts

```diff
@@ -1,63 +1,150 @@
 import yargs from 'yargs';
+import * as os from 'os';
 
 /**
  * A logger that logs messages to the console.
  */
 export class Logger {
+  // Properties for ellipsis animation
+  private timer: ReturnType<typeof setTimeout>;
+  private refreshRate: number;
+  private timerSet: boolean;
+
   /**
-   * Creates a new Logger instance.
-   * @param console The console to log to.
+   * Creates a new Logger instance. Injecting stdout for testing.
+   * @param stdout The stream to write logs to.
    * @param minimumLogLevel The minimum log level to log.
    */
   constructor(
-    private readonly console: Console = global.console,
+    private readonly stdout = process.stdout,
     private readonly minimumLogLevel: LogLevel = LogLevel.INFO
-  ) {}
+  ) {
+    this.refreshRate = 500; // every 0.5 seconds
+  }
 
   /**
    * Logs a message to the console.
    */
-  async log(message: string, level: LogLevel = LogLevel.INFO) {
+  log(message: string, level: LogLevel = LogLevel.INFO) {
     const toLogMessage = level <= this.minimumLogLevel;
 
     if (!toLogMessage) {
       return;
     }
 
-    const logMessage = `[${
-      LogLevel[level]
-    }] ${new Date().toISOString()}: ${message}`;
-    this.console.log(logMessage);
+    const logMessage =
+      this.minimumLogLevel === LogLevel.DEBUG
+        ? `[${LogLevel[level]}] ${new Date().toISOString()}: ${message}`
+        : message;
+    this.stdout.write(logMessage + os.EOL);
+  }
+
+  /**
+   * Logs a message with animated ellipsis
+   */
+  async withEllipsis(message: string, callback: () => Promise<void>) {
+    this.startAnimatingEllipsis(message);
+    await callback();
+    this.stopAnimatingEllipsis(message);
+  }
+
+  /**
+   * Start animating ellipsis at the end of a log message.
+   */
+  private startAnimatingEllipsis(message: string) {
+    if (!this.isTTY()) {
+      this.log(message, LogLevel.INFO);
+      return;
+    }
+
+    if (this.timerSet) {
+      throw new Error(
+        'Timer is already set to animate ellipsis, stop the current running timer before starting a new one.'
+      );
+    }
```

Comment:
we can do that or increment/decrement counter to keep track of nested calls and remove elipsis if counter drops to zero.
(this can be deferred until we have real need for this).

Reply:
My preference is only do this if/when needed

Reply:
Though the way this is implemented, the time this is called is for another message to have animated ellipsis so not sure if the need is there for now until this is changed.